core/src/main/java/org/apache/carbondata/core/datastore/SegmentTaskIndexStore.java

```diff
@@ -233,8 +235,7 @@ private SegmentTaskIndexWrapper loadAndGetTaskIdToSegmentsMap(
                   taskIdToTableBlockInfoMap.entrySet().iterator();
               long requiredSize =
                   calculateRequiredSize(taskIdToTableBlockInfoMap, absoluteTableIdentifier);
-              segmentTaskIndexWrapper
-                  .setMemorySize(requiredSize + segmentTaskIndexWrapper.getMemorySize());
+
               boolean isAddedToLruCache =
```

Comment:
Putting the segment cache object prior to loading the segment b-tree could 
allow dirty read in case of concurrent query.
Better to add the cache object after finishing the segment load.